core/blockchain.go

```diff
@@ -1767,18 +1734,19 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 			// is the fork block and that the conversion needs to be marked at started.
 			if !bc.stateCache.InTransition() && !bc.stateCache.Transitioned() {
 				bc.stateCache.StartVerkleTransition(parent.Root, emptyVerkleRoot, bc.Config(), bc.Config().PragueTime, parent.Root)
+				bc.stateCache.SetLastMerkleRoot(parent.Root)
 			}
 		} else {
 			// If the verkle activation time hasn't started, declare it as "not started".
 			// This is so that if the miner activates the conversion, the insertion happens
 			// in the correct mode.
 			bc.stateCache.InitTransitionStatus(false, false)
 		}
-		if parent.Number.Uint64() == conversionBlock {
-			bc.StartVerkleTransition(parent.Root, emptyVerkleRoot, bc.Config(), &parent.Time, parent.Root)
-			bc.stateCache.SetLastMerkleRoot(parent.Root)
+		stateRoot := parent.Root
+		if block.Header().Number.Uint64() == 4702178 {
+			stateRoot = common.HexToHash("0x00")
 		}
```

Comment:
So this was a very ugly rabbit hole due to a bug in the way our previous replay data was exported.

In the previous setup the first block to be replayed was 4702177 and unfortunately that block had a snapshot in the state. We need to force any second (i.e: previously 4702178) or further block to always read from the tree and **not** use snapshots.

Maybe the new replay data wasn't corrupted in that way and we can remove this. But it's worth checking that `state.New(stateRoot, ...)` in the line below *never* finds a snapshot for the second-and-further replayed blocks, it must always read from the tree. If that isn't the case, we might be replaying using snapshots prepopulated by the original execution that created the exported chain and not the replayed blocks.